### PR TITLE
add error message for dolt commit failures

### PR DESCRIPTION
**Pull request by stephkyou**
adds additional error message when the dolt commit editor fails to open

fixes: https://github.com/dolthub/dolt/issues/5060

### CODE REVIEW

File: integration-tests/bats/merge.bats [hunk 1]
```diff
@@ -997,3 +997,14 @@ SQL
     [ $status -eq 0 ]
     [[ "$output" =~ "2 tables changed, 3 rows added(+), 1 rows modified(*), 1 rows deleted(-)" ]] || false
 }
+
+@test "merge: failed to open editor" {
+  dolt config --add core.editor "vi --broken"
```
Comment (timsehn) [inline integration-tests/bats/merge.bats:1002]: This should be `export EDITOR="vi --broken"`

File: go/cmd/dolt/commands/commit.go [hunk 2]
```diff
@@ -322,12 +316,22 @@ func getCommitMessageFromEditor(ctx context.Context, dEnv *env.DoltEnv, suggeste
 	})
 
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("Failed to open commit editor")
```
[the remark this answers is not in the document]
Comment (fulghum) [reply go/cmd/dolt/commands/commit.go:319]: Actually... we might be able to make this error message even more helpful/actionable... Maybe we could report the command we tried to run, or ask users to check their `EDITOR` env var or their Dolt config to make sure it's a valid editor? 

Git typically does a really, really good job giving actionable messages to help users know what step to take next. A good pattern for us to strive for, too.

**Review by fulghum — COMMENTED** [review]
Nice fix @stephkyou! Code looks correct, just some comments on how we could make the error message even easier for users to understand and to take action on. 

Error messages can have **such** a big impact on a customer's experience with a tool, so I think it's worth trying out a few ideas to see if we can polish it more. In general, having complete details (e.g. perhaps showing the exact editor command we tried to run?) and suggesting a next step for users (e.g. checking the EDITOR env var and Dolt config editor setting) goes a long way for helping someone easily deal with an error.

**Review by fulghum — APPROVED** [review]
Beautiful error message! Very nice detail in there that will be helpful to the next customer who hits this. 😁 Nice work!

**Pull request merged**